### PR TITLE
Make spec character rendering a bit cleaner

### PR DESCRIPTION
**Pull request by heinrich5991**

### CODE REVIEW

**Comment by heinrich5991** [comment]
WIP, doesn't render nameplates yet.

File: src/game/server/player.cpp [hunk 1]
```diff
@@ -412,15 +412,11 @@ void CPlayer::Snap(int SnappingClient)
 		pRaceInfo->m_RaceStartTick = m_pCharacter->m_StartTime;
 	}
 
-	if(m_Paused == PAUSE_SPEC && m_pCharacter && m_pCharacter->Core())
+	if(m_pCharacter && m_pCharacter->Core() && m_pCharacter->IsPaused())
```
Comment (Learath2) [inline src/game/server/player.cpp:415]: Don't need to check `m_pCharacter->Core()`, can't possibly not exist.

File: src/game/client/components/players.cpp [hunk 2]
```diff
@@ -680,57 +652,53 @@ void CPlayers::OnRender()
...
+			if(p % 3 != 0 && !m_pClient->m_Snap.m_aCharacters[i].m_Active)
 				continue;
 
-			const void *pPrevInfo = Client()->SnapFindItem(IClient::SNAP_PREV, NETOBJTYPE_PLAYERINFO, i);
-			const void *pInfo = Client()->SnapFindItem(IClient::SNAP_CURRENT, NETOBJTYPE_PLAYERINFO, i);
-
-			if(pPrevInfo && pInfo)
+			if(p % 3 == 0)
 			{
+				if(p < 3)
```
Comment (Learath2) [inline src/game/client/components/players.cpp:672]: What is this for? `p > 0 && p % 3 == 0 && p < 3` <=> `p == 0`

File: src/game/client/components/players.cpp [hunk 3]
```diff
@@ -680,57 +652,53 @@ void CPlayers::OnRender()
 			}
 		}
 	}
+	m_RenderInfoSpec.m_Texture = m_pClient->m_pSkins->Get(m_pClient->m_pSkins->Find("x_spec"))->m_OrgTexture;
```
Comment (Learath2) [inline src/game/client/components/players.cpp:655]: You could look this up once `OnInit`. Probably not significant but another linear lookup we don't need while rendering.

File: src/game/client/components/nameplates.cpp [hunk 4]
```diff
@@ -172,18 +168,25 @@ void CNamePlates::OnRender()
...
+		const CNetObj_PlayerInfo *pInfo = (CNetObj_PlayerInfo *)pInfoRaw;
+		if(!pInfo)
+		{
 			continue;
+		}
 
-		const void *pInfo = Client()->SnapFindItem(IClient::SNAP_CURRENT, NETOBJTYPE_PLAYERINFO, i);
+		if(m_pClient->m_aClients[i].m_SpecCharPresent)
+		{
+			RenderNameplatePos(m_pClient->m_aClients[i].m_SpecChar, pInfo, 0.4f);
```
Comment (Learath2) [inline src/game/client/components/nameplates.cpp:180]: Should this use `g_Config.m_ClShowOthersAlpha`? I think it's better like this but it's a change from the old behaviour.

**Comment by def-** [comment]
Thanks for all the cleanups!
Tested locally, looks all good.
bors r+

File: src/game/client/components/players.cpp (hunk 2 above)
Comment (heinrich5991) [reply src/game/client/components/players.cpp:672]: This was for rendering the hook, but we don't do that for the spec tees, wanted to keep the structure.

File: src/game/client/components/nameplates.cpp (hunk 4 above)
Comment (heinrich5991) [reply src/game/client/components/nameplates.cpp:180]: This was not reusing the other config variable on purpose.

**Pull request merged**